.github/workflows/ci.yml

```
@@ -328,13 +346,26 @@ jobs:
 
       - name: Test
         run: |
-          ./mithril-end-to-end -vvv \
-            --bin-directory ./bin \
-            --work-directory=./artifacts \
-            --devnet-scripts-directory=./mithril-test-lab/mithril-devnet \
-            --mithril-era=${{ matrix.era }} \
-            --cardano-node-version ${{ matrix.cardano_node_version }} \
-            --cardano-hard-fork-latest-era-at-epoch ${{ matrix.hard_fork_latest_era_at_epoch }} ${{ matrix.extra_args }}
+          if [[ "${{ matrix.next_era }}" == "" ]]; then
+            # If there is no next era, we don't need to specify it with '--mithril-next-era'
+            ./mithril-end-to-end -vvv \
+              --bin-directory ./bin \
+              --work-directory=./artifacts \
+              --devnet-scripts-directory=./mithril-test-lab/mithril-devnet \
+              --mithril-era=${{ matrix.era }} \
+              --cardano-node-version ${{ matrix.cardano_node_version }} \
+              --cardano-hard-fork-latest-era-at-epoch ${{ matrix.hard_fork_latest_era_at_epoch }} ${{ matrix.extra_args }}
+          else
+            # If there is a next era, we need to specify it with '--mithril-next-era'
+            ./mithril-end-to-end -vvv \
+              --bin-directory ./bin \
+              --work-directory=./artifacts \
+              --devnet-scripts-directory=./mithril-test-lab/mithril-devnet \
+              --mithril-era=${{ matrix.era }} \
+              --mithril-next-era=${{ matrix.next_era }} \
+              --cardano-node-version ${{ matrix.cardano_node_version }} \
+              --cardano-hard-fork-latest-era-at-epoch ${{ matrix.hard_fork_latest_era_at_epoch }} ${{ matrix.extra_args }}
+          fi
```

I’m not entirely sure about the syntax, but it might be worth unifying the base command and only adding the argument if necessary to avoid duplication. Something like that:

```suggestion
          CMD="./mithril-end-to-end -vvv \
            --bin-directory ./bin \
            --work-directory=./artifacts \
            --devnet-scripts-directory=./mithril-test-lab/mithril-devnet \
            --mithril-era=${{ matrix.era }} \
            --cardano-node-version ${{ matrix.cardano_node_version }} \
            --cardano-hard-fork-latest-era-at-epoch ${{ matrix.hard_fork_latest_era_at_epoch }} ${{ matrix.extra_args }}"

          # If there is a next era, we need to specify it with '--mithril-next-era'
          if [[ "${{ matrix.next_era }}" != "" ]]; then
            CMD+=" --mithril-next-era=${{ matrix.next_era }}"
          fi
          
          $CMD
```

---

I tried something in https://github.com/input-output-hk/mithril/pull/1974/commits/2403723e3cd7f1dfa652c20adcfdde2ccff2618b :+1: